### PR TITLE
[2.5] Ensure system and default projects get created

### PR DESCRIPTION
**Pull request by thedadams**
Original PR:
https://github.com/rancher/rancher/pull/34860

Issue:
https://github.com/rancher/rancher/issues/34952

### CODE REVIEW

File: pkg/controllers/managementuser/rbac/namespace_handler.go [hunk 1]
```diff
@@ -134,6 +159,27 @@ func (n *nsLifecycle) assignToInitialProject(ns *v1.Namespace) error {
...
+		return "", fmt.Errorf("cluster [%s] contains more than 1 [%s] project", n.m.clusterName, projectpkg.System)
+	}
+	if projects[0] == nil {
+		return "", nil
+	}
+	return projects[0].Name, nil
+}
+
+func IsFleetNamespace(ns *v1.Namespace) bool {
+	return ns.Name == "fleet-local" || ns.Name == "fleet-default" || ns.Name == "cattle-fleet-clusters-system" || ns.Labels["fleet.cattle.io/managed"] == "true"
```
Comment (StrongMonkey) [inline pkg/controllers/managementuser/rbac/namespace_handler.go:180]: This needs to be `fleet-clusters-system`. In 2.5 we haven't change the fleet system namespace from `fleet-system` to `cattle-fleet-system`. (https://github.com/rancher/fleet/issues/395)

File: pkg/controllers/managementuser/rbac/namespace_handler.go [hunk 2]
```diff
@@ -87,6 +88,30 @@ func (n *nsLifecycle) Remove(obj *v1.Namespace) (runtime.Object, error) {
 }
 
 func (n *nsLifecycle) syncNS(obj *v1.Namespace) (bool, error) {
+	// add fleet namespace to system project
+	if IsFleetNamespace(obj) &&
```
Comment (StrongMonkey) [inline pkg/controllers/managementuser/rbac/namespace_handler.go:92]: I would be ok with this, but we are actually backporting https://github.com/rancher/rancher/issues/31209 to 2.5 then we should open a new issue for 2.5 and let qa verify it.

File: pkg/controllers/managementuser/rbac/namespace_handler.go (hunk 1 above)
Comment (thedadams) [reply pkg/controllers/managementuser/rbac/namespace_handler.go]: Updated.

File: pkg/controllers/managementuser/rbac/namespace_handler.go (hunk 2 above)
Comment (thedadams) [reply pkg/controllers/managementuser/rbac/namespace_handler.go]: Removed.

File: pkg/app/utils.go [hunk 3]
```diff
@@ -163,3 +164,8 @@ func DeleteApp(mgmtAppClient projv3.AppInterface, projectID, appName string) err
 
 	return nil
 }
+
+func isKnownSystemNamespace(ns *v1.Namespace) bool {
+	return slice.ContainsString(strings.Split(settings.SystemNamespaces.Get(), ","), ns.Name, nil) ||
+		rbac.IsFleetNamespace(ns)
```
Comment (StrongMonkey) [inline pkg/app/utils.go:170]: Also probably you don't need `rbac.IsFleetNamespace(ns)` too since they are not added to system namespace anymore

Comment (thedadams) [reply pkg/app/utils.go]: Removed.

**Review by StrongMonkey — DISMISSED** [review]

File: pkg/rancher/migrations.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	rancherversion "github.com/rancher/rancher/pkg/version"
+	controllerv1 "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	v1 "k8s.io/api/core/v1"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	cattleNamespace                           = "cattle-system"
+	forceUpgradeLogoutConfig                  = "forceupgradelogout"
```
Comment (snasovich) [inline pkg/rancher/migrations.go:15]: `forceUpgradeLogoutConfig` and `rancherVersionKey` are not needed here.

**Review by snasovich — APPROVED** [review]
LGTM, one minor comment that would be nice to address

File: pkg/rancher/migrations.go (hunk 4 above)
Comment (thedadams) [reply pkg/rancher/migrations.go]: Done.

**Review by snasovich — APPROVED** [review]

**Pull request merged**